clients/src/main/java/org/apache/kafka/clients/consumer/RangeAssignor.java

```diff
@@ -48,27 +71,30 @@ public String name() {
         return "range";
     }
 
-    private Map<String, List<String>> consumersPerTopic(Map<String, Subscription> consumerMetadata) {
-        Map<String, List<String>> res = new HashMap<>();
+    private Map<String, List<MemberInfo>> consumersPerTopic(Map<String, Subscription> consumerMetadata) {
+        Map<String, List<MemberInfo>> topicToConsumers = new HashMap<>();
         for (Map.Entry<String, Subscription> subscriptionEntry : consumerMetadata.entrySet()) {
             String consumerId = subscriptionEntry.getKey();
-            for (String topic : subscriptionEntry.getValue().topics())
-                put(res, topic, consumerId);
+            MemberInfo memberInfo = new MemberInfo(consumerId, subscriptionEntry.getValue().groupInstanceId());
```

Comment:
Are these members guaranteed to be sorted by `member.id` already? I wasn't able to see how this change is going to result in a stable assignment. Sorry if I'm missing something obvious.